search-app/docs/api.md

```diff
@@ -5405,3 +5409,101 @@ Example un-healthy response body:
   }
 }
 ```
+### <a name="associate-any-concepts"></a> Associate any concepts
+
+A new association API is developed to achieve the goal of being able to associate a concept of any type, with or without revision, to one or more other concepts of any type, with or without revisions. The new association API also allows the associations to include an optional data payload, whose purpose is to describe the association itself. Associations which do not initially have association data payload, may have it added through association update. A concept can only be associated with another concept either with or without revisions, not both. A concept can not be associated to itself, even with different revisions. It's worth noting that associations between collections and services/tools/variables can not be made through the new association API because these associations require different business rules and there exist some complexites with the way how certain fields of the association entries are constructed. Until we resolve these issues, these existing associations will continue to be made through the existing association API. 
+ 
+#### <a name="concept-associations"></a> Concept associations 
+
+A concept, with optional revision id, can be associated to one or more other concepts, with optional revision ids and data payloads.
+
+ Example
+
+ ```
+ curl -XPOST -H "Authorization: Bearer XXXXX" -H "Content-Type:application/json"
+      "%CMR-ENDPOINT%/associate/SE1200000006-PROV1/3"
+      -d '[{"concept_id": "V1200000008-PROV1", "revision_id": 1},
+           {"concept_id": "C1200000005-PROV1", "data": {"field": "value", "another_field": {"XYZ": "ZYX"}}}]'
```

Comment:
The API doc says `It's worth noting that associations between collections and services/tools/variables can not be made through the new association API`, but the example is showing the new association API with collection. Also the example response does not match the example search.

Reply:
So in this case, the Service entry is being associated with a variable and a collection which is allowed. What would not be allowed is say, associate a collection -> variable. Something like 
` "%CMR-ENDPOINT%/associate/C1200000005-PROV1/3"
      -d '[{"concept_id": "V1200000008-PROV1", "revision_id": 1},`

Reply:
The example is the association between collection and service entry, not with service. they are different concept types. I will double check the response.

Reply:
OK. It is not very clean. I see it now. The response could be updated, not showing the collection concept is quite confusing and I don't think it is acceptable.

Reply:
CMR-8789 is filed.